### PR TITLE
Pin Assign Opt in the Add Constraint file form

### PR DESCRIPTION
**Pull request by zaryabkhan-rs**
> ### Motivate of the pull request
> - [x] To address an existing issue. RG-17
> - [ ] Breaking new feature. If so, please describe details in the description part.

> ### Describe the technical details
> #### What is currently done? (Provide issue link if applicable)
>- Added pin assignment option in Add_Constraint_form dialog from New Project Window.
>- Added pin assignment option in Add Source dialog from Source tab on GUI. 
>- Selecting Pin Assignment settings from Add_Constraint_form/Add Source dialog will be saved and pass to placement settings after completing new project flow. 

> #### What does this pull request change?
>  This PR improves in the following aspects:  
>- Pin Assignment Method Selection from different dialogs.

> 
> 

> ### Testing Steps:
>- create_design project_1
>- Select pin option from add_constraint_form and check on the placement setting either the selection is set or not
>- pin_loc_assign_method random/in_define_order  -->  change the pin assign selection from console.

> ### Which part of the code base require a change
> 
> - [ ] Library: <Specify the library name>
> - [ ] Plug-in: <Specify the plugin name>
> - [x] Engine
> - [ ] Documentation
> - [x] Regression tests
> - [x] Continous Integration (CI) scripts

> ### Impact of the pull request

> - [x] Added new options to GUI for Add Constraint Form

> ### Update Screenshots

image
image

### CODE REVIEW

File: src/NewProject/add_constraints_form.cpp [hunk 1]
```diff
@@ -22,6 +26,17 @@ addConstraintsForm::addConstraintsForm(QWidget *parent)
 
   ui->m_ckkBoxCopy->setText(tr("Copy sources into project."));
   ui->m_ckkBoxCopy->setCheckState(Qt::CheckState::Checked);
+
+  Compiler *compiler = GlobalSession->GetCompiler();
+  if (compiler->PinAssignOpts() == Compiler::PinAssignOpt::Random) {
+    ui->select_random->setChecked(true);
+  } else
```
Comment (KochynVolodymyr) [inline src/NewProject/add_constraints_form.cpp:33]: please put {} here or remove for **if** - for consistency

File: src/ProjNavigator/add_file_form.cpp [hunk 2]
```diff
@@ -19,6 +22,15 @@ AddFileForm::AddFileForm(QWidget *parent)
   box->setContentsMargins(0, 0, 0, 0);
   box->setSpacing(0);
   ui->m_frame->setLayout(box);
+  Compiler *compiler = GlobalSession->GetCompiler();
+  if (compiler->PinAssignOpts() == Compiler::PinAssignOpt::Random) {
+    ui->select_random->setChecked(true);
+  } else
```
Comment (KochynVolodymyr) [inline src/ProjNavigator/add_file_form.cpp:28]: same here

File: src/NewProject/add_constraints_form.cpp (hunk 1 above)
Comment (zaryabkhan-rs) [reply src/NewProject/add_constraints_form.cpp:33]: Okay

File: src/ProjNavigator/add_file_form.cpp (hunk 2 above)
Comment (zaryabkhan-rs) [reply src/ProjNavigator/add_file_form.cpp:28]: okay

Comment (zaryabkhan-rs) [reply src/ProjNavigator/add_file_form.cpp]: done

File: src/NewProject/add_constraints_form.cpp (hunk 1 above)
Comment (zaryabkhan-rs) [reply src/NewProject/add_constraints_form.cpp]: Done.

File: src/NewProject/add_constraints_form.cpp [hunk 3]
```diff
@@ -22,6 +26,17 @@ addConstraintsForm::addConstraintsForm(QWidget *parent)
...
   ui->m_ckkBoxCopy->setText(tr("Copy sources into project."));
   ui->m_ckkBoxCopy->setCheckState(Qt::CheckState::Checked);
+
+  Compiler *compiler = GlobalSession->GetCompiler();
+  if (compiler->PinAssignOpts() == Compiler::PinAssignOpt::Random)
+    ui->select_random->setChecked(true);
+  else
+    ui->select_defineOrder->setChecked(true);
+
+  connect(ui->select_defineOrder, &QRadioButton::clicked, this,
```
Comment (KochynVolodymyr) [inline src/NewProject/add_constraints_form.cpp:36]: critical issue is here. User can press Cancel but the selection will take affect. User selection need to buffer somewhere and apply to compiler only after user press Ok.

File: src/ProjNavigator/add_file_form.cpp [hunk 4]
```diff
@@ -19,6 +22,15 @@ AddFileForm::AddFileForm(QWidget *parent)
   box->setContentsMargins(0, 0, 0, 0);
   box->setSpacing(0);
   ui->m_frame->setLayout(box);
+  Compiler *compiler = GlobalSession->GetCompiler();
+  if (compiler->PinAssignOpts() == Compiler::PinAssignOpt::Random)
+    ui->select_random->setChecked(true);
+  else
+    ui->select_defineOrder->setChecked(true);
+  connect(ui->select_defineOrder, &QRadioButton::clicked, this,
```
Comment (KochynVolodymyr) [inline src/ProjNavigator/add_file_form.cpp:30]: same here

File: src/NewProject/add_constraints_form.cpp (hunk 3 above)
Comment (zaryabkhan-rs) [reply src/NewProject/add_constraints_form.cpp:36]: Okay

**Review by ravic-rs — CHANGES_REQUESTED** [review]
Please fix the CI test failures.

File: src/ProjNavigator/add_file_form.cpp (hunk 4 above)
Comment (zaryabkhan-rs) [reply src/ProjNavigator/add_file_form.cpp]: Fixed

**Comment by zaryabkhan-rs** [comment]
> Please fix the CI test failures.

Fixed.

File: src/NewProject/add_constraints_form.h [hunk 5]
```diff
@@ -20,6 +20,7 @@ class addConstraintsForm : public QWidget {
 
   QList<filedata> getFileData();
   bool IsCopySource();
+  bool IsRandom();
```
Comment (KochynVolodymyr) [inline src/NewProject/add_constraints_form.h:23]: bool IsRandom() const;

File: src/ProjNavigator/add_file_form.h [hunk 6]
```diff
@@ -25,6 +25,7 @@ class AddFileForm : public QWidget {
   QString getFileSet() const;
   QList<filedata> getFileData() const;
   bool IsCopySource() const;
+  bool IsRandom();
```
Comment (KochynVolodymyr) [inline src/ProjNavigator/add_file_form.h:28]: bool IsRandom() const;

File: src/NewProject/add_constraints_form.h (hunk 5 above)
Comment (zaryabkhan-rs) [reply src/NewProject/add_constraints_form.h]: Fixed.

File: src/ProjNavigator/add_file_form.h (hunk 6 above)
Comment (zaryabkhan-rs) [reply src/ProjNavigator/add_file_form.h]: Fixed.

**Review by KochynVolodymyr — APPROVED** [review]

**Review by ravic-rs — APPROVED** [review]
Looks good.

**Pull request merged**